db/migrate/20181016140921_migrate_orch_stacks_to_have_ownership_concept.rb

```diff
@@ -0,0 +1,79 @@
+class MigrateOrchStacksToHaveOwnershipConcept < ActiveRecord::Migration[5.0]
+  class ExtManagementSystem < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+
+    def tenant_identity
+      User.super_admin
+    end
+  end
+
+  class Service < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+    belongs_to :evm_owner, :class_name => "MigrateOrchStacksToHaveOwnershipConcept::User"
+
+    def tenant_identity
+      user = evm_owner
+      user = User.super_admin if user.nil? || !user.miq_group_ids.include?(miq_group_id)
+      user
+    end
+  end
+
+  class MiqGroup < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+    belongs_to :tenant, :class_name => "MigrateOrchStacksToHaveOwnershipConcept::Tenant"
+  end
+
+  class User < ApplicationRecord
+    has_and_belongs_to_many :miq_groups, :class_name => "MigrateOrchStacksToHaveOwnershipConcept::MiqGroup"
+    belongs_to :current_group, :class_name => "MigrateOrchStacksToHaveOwnershipConcept::MiqGroup"
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+
+    def self.super_admin
+      find_by(:userid => "admin")
+    end
+
+    def current_tenant
+      current_group.tenant
+    end
+  end
+
+  class ServiceResource < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+    belongs_to :service, :class_name => "MigrateOrchStacksToHaveOwnershipConcept::Service"
+  end
+
+  class Tenant < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class OrchestrationStack < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+    has_many :service_resources, :as => :resource, :class_name => "MigrateOrchStacksToHaveOwnershipConcept::ServiceResource"
+    has_many :direct_services, :through => :service_resources, :source => :service # does this also need the class_name?
+    belongs_to :ext_management_system, :foreign_key => :ems_id
+
+    def service
+      direct_services.first
+    end
+  end
+
+  def up
+    say_with_time("Migrating existing orchestration stacks to have direct owners, groups, tenant") do
+      OrchestrationStack.all.each do |stack|
+        user = if stack.service.present?
+                 stack.service.tenant_identity
+               elsif stack.ems_id != nil
+                 stack.ext_management_system.tenant_identity
+               else
+                 User.super_admin
+               end
+        stack.update_attributes(:evm_owner_id => user.id, :tenant_id => user.current_tenant.id, :miq_group_id => user.current_group.id)
```

Comment:
is it possible for `user` or `user.current_tenant` or `user.current_group` to be nil?

Reply:
in this test, yes `current_group is nil` - but in the real world. no

Reply:
Basically, if a user exists in this system, then they have an `miq_group`.
And if they have an `miq_group`, then they have an `current_group`. (code enforces that)

This is good as is